helix-rest/src/main/java/org/apache/helix/rest/server/service/PropertyStoreService.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.helix.rest.server.service;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.client.HelixZkClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class PropertyStoreService {
```

Comment:
I see what you're doing, but I don't think a whole class is necessary for this purpose. See my comment above.

Reply:
Please see changes at the latest commit, if you feel without the class the code is cleaner we can go with that route

Reply:
We should try to avoid using ZkClient directly to read and write. I did see your comment. I think it's possible to do this without this new class and using getPropertyStore().

- You could create a private object using new ZkHelixPropertyStore<ZNRecord>(). If you think this is hard, you could use ZkBaseDataAccessor<ZNRecord> as the last resort.
- If ZNode we are reading is not in the ZNRecord format, we could catch that exception and do appropriate processing (say, include it in the mapField of Json).

The second step actually would need some work. You could define a dummy serializer that simply reads the byte array and converts it to a String value in the Json. 

Does this make sense? Let me know if you have any questions.

Reply:
Let's use ZKClient directly at this PR, it's just simple. My latest commit the Service class is no longer used at all. I've taken a look at the ZkHelixPropertyStore class which provides a lot of overheads that are not needed at all in Helix API service. 

Ideally, I will prefer to use ZkBaseDataAccessor with the Non-ZnRecord custom serializer support(add a getBaseDataAccesor method in abstract class)  (will be my next PR).

Reply:
@i3wangyi The reason to avoid using ZkClient directly is because we need to allow for the possibility of redirection at the base data accessor level. 
Consider a case where Helix uses multiple metadata stores in the future?

Reply:
I couldn't agree with you on the long long future plan, without detailed migration plans it's hard to convince ppl to give up the usage of a public class (zkClient) plus the usage is only with Helix Rest. The introduced overhead from ZKPropertyStore class where it registers cache/ cache update thread, ..., will make it more complicated to debug than the simple ZKClient method. I'm afraid the usage becomes a magic blackbox when the return value is NULL you don't know it means the value is empty or not a znRecord record.
I will prefer adding a todo here once we have the ZkBaseDataAccessor with the Non-ZnRecord custom serializer support, I could use the baseDataAccessror than the ZkClient.

Reply:
Let us discuss using ZkBaseDataAccessor so that we could achieve this without using ZkClient directly here. I think it's possible. Let us sync in person.

Reply:
Does it mean the PR will need to wait on the baseDataAccesor's non-znRecord methods ready? It's likely to be so. Any ideas to expedite and run things in parallel?

Reply:
We could have the first PR to be the addition of the "custom" ZkClient that just reads byte[]. This will be the building block for all of the other PRs. 

That shouldn't be too complicated, and upon completion of that, you could rebase other PRs to be able to leverage that. Does that sound good to you?